tools/testsys/src/aws_resources.rs

```diff
@@ -173,23 +208,29 @@ impl AwsK8s {
                 conflicts_with: None,
                 agent: Agent {
                     name: "eks-provider".to_string(),
-                    image: testsys_images.eks_resource.clone(),
-                    pull_secret: testsys_images.secret.clone(),
+                    image: testsys_images
+                        .eks_resource_agent_image
+                        .as_ref()
+                        .cloned()
+                        // This is a safe unwrap because `testsys_images` has a default value set
+                        // for `eks_resource_agent`.
+                        .unwrap(),
```

Comment:
nit: this is the same info but more useful to a developer that sees the theoretical panic:
```suggestion
                        .expect("missing default image for eks resource agent"),
```